services/service-runner/app/service/service-runner.ts

```diff
@@ -288,84 +287,138 @@ export async function runServiceFromPull(
+                );
+              }
+
+              await uploadLogs(workItem, [retryMessage, debugInfo, ...stdOut.logStrArr]);
+
               clearTimeout(timeout);
-              const errorEntries = await _getErrorInfo(status, catalogDir, workItemLogger);
-              const errorMessage = `${serviceName}: ${errorEntries.error}`;
-              const errorLevel = errorEntries.level;
-              const errorCategory = errorEntries.category;
-              if (errorCategory) {
-                resolve({ error: errorMessage, errorLevel, errorCategory });
+
+              if (status.status === 'Success') {
+                workItemLogger.debug('Getting STAC catalogs');
+                const catalogs = await _getStacCatalogs(catalogDir);
+                resolve({ batchCatalogs: catalogs });
+              } else {
+                const errorEntries = await _getErrorInfo(status, catalogDir, workItemLogger);
+                const errorMessage = `${serviceName}: ${errorEntries.error}`;
+                const errorLevel = errorEntries.level;
+                const errorCategory = errorEntries.category;
+
+                if (errorCategory) {
+                  resolve({ error: errorMessage, errorLevel, errorCategory });
+                } else if (status.code === 500) {
+                  // The k8s client hit an error, the worker did not return an error
+                  // In this scenario we want to retry internally rather than immediately fail
+                  workItemLogger.error('K8s hit an internal error, will attempt to retry until retries are exhausted');
+                  resolve({ retryable: true });
+                } else {
+                  resolve({ error: errorMessage, errorLevel });
+                }
               }
-              resolve({ error: errorMessage, errorLevel });
+            } catch (e) {
+              clearTimeout(timeout);
+              workItemLogger.error('Caught exception while executing work:');
+              workItemLogger.error(e);
+              resolve({ error, errorLevel: 'error' });
             }
-          } catch (e) {
-            workItemLogger.error('Unable to upload logs. Caught exception:');
-            workItemLogger.error(e);
-            resolve({ error, errorLevel: 'error' });
-          }
-        },
-      ).catch((e) => {
-        clearTimeout(timeout);
-        workItemLogger.error('Kubernetes client exec caught exception:');
-        workItemLogger.error(e);
-        resolve({ error, errorLevel: 'error' });
+          },
+        ).catch((e) => {
+          clearTimeout(timeout);
+          workItemLogger.error('Kubernetes client exec caught exception:');
+          workItemLogger.error(e);
+          resolve({ error, errorLevel: 'error' });
+        });
       });
-    });
+
+      if ('retryable' in result) {
+        workItemLogger.debug(`Retryable error encountered (attempt ${retryCount + 1} of ${maxRetries})`);
```

Comment:
nit: Move line 409 up above this line, so you can log `...attempt ${retryCount}...`, rather than +1.